components/camel-swagger-java/src/main/java/org/apache/camel/swagger/RestSwaggerSupport.java

```diff
@@ -345,4 +353,36 @@ private static void setupCorsHeaders(RestApiResponseAdapter response, Map<String
         response.setHeader("Access-Control-Max-Age", maxAge);
     }
 
+    private void setupXForwardedHeaders(Swagger swagger, Map<String, Object> headers) {
+
+        String host = (String) headers.get(HEADER_HOST);
+        if(host != null && host.length() > 0) {
```

Comment:
nitpick, could use:

```java
if (ObjectHelper.isNotEmpty(host)) {
```

Reply:
it should work, the prefix could also come with "/" at the end, I will apply this suggestion.